### ISSUE TITLE
False positive RS1024 when creating HashSet with `SymbolEqualityComparer`

### ISSUE THREAD

**Issue by huoyaoyuan** [dotnet/roslyn-analyzers#5804]
### Analyzer

**Diagnostic ID**: RS1024: `Compare symbols correctly`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysisAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.Analyzers)

**Version**: 3.3.2 (Carried by Roslyn 4.0.1)

### Describe the bug

Duplicate of #4568

### Steps To Reproduce

```csharp
void Main()
{
    var usedTypes = new HashSet<ITypeSymbol>(SymbolEqualityComparer.Default);
    var usedSymbols = new HashSet<ISymbol>(SymbolEqualityComparer.Default);
}
```
Reproduces in minimal project.

### Expected behavior

No warning.

### Actual behavior

Warning RS1024

## Additional context

This is a regression from 3.0.0, the version carried by Roslyn 3.9.0.
https://github.com/dotnet/roslyn-analyzers/blob/f471d3381584f10f9908432e0b2b2b8ef07a0aa6/src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/CompareSymbolsCorrectlyTests.cs#L1087-L1112
It looks like covered by unit test. Curious how it could happen.

**Comment by Youssef1313** [comment]
This is fixed in 3.3.3

**Comment by huoyaoyuan** [comment]
OK. The latest preview version (4.1.0-1.final) hasn't been updated yet. I'll close this when I verify in published version.

**Comment by Flash0ver** [comment]
@Youssef1313, the fix you've described only accounts for `SymbolEqualityComparer.Default`, but not for custom `EqualityComparer<ISymbol>` implementations, does it?

**Comment by Youssef1313** [comment]
@Flash0ver Not sure, but probably. This would be tracked by https://github.com/dotnet/roslyn-analyzers/issues/5715.

**Comment by Flash0ver** [comment]
@Youssef1313, thanks for your magnificently fast feedback - as I'm encountering that issue right now.

**Comment by Youssef1313** [comment]
@Flash0ver I'll see if I can work on a fix soon.

**Comment by Youssef1313** [comment]
@huoyaoyuan Can you please verify with `4.1.0-3.final`?

**Comment by huoyaoyuan** [comment]
Yes it's fixed.

**Comment by Flash0ver** [comment]
Sorry for my late response - and thank you for the super quick fix. I appreciate it!

**Comment by eiriktsarpalis** [comment]
We're still seeing this false positive in the dotnet/runtime repo, cf. https://github.com/dotnet/runtime/pull/86616#discussion_r1202360928

**Comment by Youssef1313** [comment]
@eiriktsarpalis It should work if you use the latest version of the analyzers package. I added a comment to the PR for my guess as to why it's not working in your case.
